src/main/java/com/google/gwt/dom/client/InputElement.java

```diff
@@ -381,6 +384,19 @@ public final void setValue(String value) {
   }-*/;
 
   /**
+   * @Mock
+   * 
+   * Set value in native element (like Selenium 'sendKeys').
+   * The new value will be available in calls to getValue().
+   * N.B. Will not trigger ValueChange events until focus lost.
+   */
+  public final void setNewValue(String value) {
+      focus();
+      setAttribute("mock.Value.new", value);
```

Comment:
About testing Change and FocusLost events: if I understand things right, your application might accidentally rely on the order of Change and Blur events and you want to test that your application work fine in both cases (Change-Blur, and Blur-Change). To test this, I think you can just programmtically fire these events in both orders, and these don't need special extensions to gwt-mock. I would also add a reusable utility class that encapsulates this blur-change order problem and your components can rely on that class. What do you think?